enhancements/etcd/protecting-etcd-quorum-during-control-plane-scaling.md

```diff
@@ -0,0 +1,450 @@
+
+## Summary
+
+To enable automation of Control Plane scaling activities, in particular vertical scaling of the Control Plane Machines,
+we must implement a mechanism that protects etcd quorum and ensures the smoothest possible transition as new etcd
+members are added and old members removed from the etcd cluster.
+
+## Motivation
+
+As Red Hat expands its managed services offerings, the ability to safely increase and decrease the capacity of an
+OpenShift Control Plane in some automated manner becomes imperative.
+
+Currently, when a cluster starts to hit capacity limits on the Control Plane, a very involved manual process is
+required to not only add new Machines to the cluster, but monitor and manage the etcd cluster to ensure that the quorum
+is preserved throughout the operation.
+
+This process is not sustainable and we must provide safety mechanisms on top of the existing etcd quorum guard to make
+this process both easier and safer.
+
+### Goals
+
+* Provide the etcd Operator with the ability to control when a Control Plane Machine is removed from the cluster
+* Allow the etcd Operator to prevent removal of etcd members until a replacement member has been promoted to a voting
+  member
+* Allow the etcd Operator to remove an etcd member from the etcd cluster before the Machine is terminated to prevent a
+  degraded etcd cluster
+* Allow an escape hatch from the protection when surging the capacity with new Control Plane Machines is unavailable
+  (for example in metal environments with limited capacity)
+
+
+### Non-Goals
+
+* Automation of scaling operations on Machines
+* Horizontal scaling of the etcd cluster and Control Plane
+* Providing these protection mechanisms when the Machine API is [not functional](#When-is-Machine-API-Functional)
+* Recovering unhealthy etcd clusters
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an operator of a managed OpenShift cluster, I want to be able to automate the scaling operations of Control Plane
+Machines without having to manually ensure the safety of the etcd cluster.
+
+#### Story 2
+
+As a developer of OpenShift, I want to implement safety mechanisms that adhere to best practices for etcd scaling
+operations to protect end users from potential quorum losses and data losses.
+
+#### Story 3
+
+As an end user of OpenShift, I want to be able to increase the size of my Control Plane without having to know the
+intricacies of etcd and protecting its quorum.
+
+### API Extensions
+
+This enhancement does not introduce any new API extensions.
+
+### Implementation Details/Notes/Constraints
+
+#### Requirements for etcd safety
+
+* The number of voting members should equal the desired number of Control Plane Machines (and this should be an odd
+  number)
+  * We will only deviate from this to add a replacement member
+  * Once the new member is added, we should remove old members as soon as possible to reduce the risk of degrading the
```

Comment:
old member or member**s** ?

Reply:
Yeah this should be singular, good catch